File: LeetCode/strings/One-Edit-Distance.js
// Given two strings s and t, determine if they are both one edit distance apart.

// Note: 

// There are 3 possiblities to satisify one edit distance apart:

// Insert a character into s to get t
// Delete a character from s to get t
// Replace a character of s to get t
// Example 1:

// Input: s = "ab", t = "acb"
// Output: true
// Explanation: We can insert 'c' into s to get t.
// Example 2:

// Input: s = "cab", t = "ad"
// Output: false
// Explanation: We cannot get t from s by only one step.
// Example 3:

// Input: s = "1203", t = "1213"
// Output: true
// Explanation: We can replace '0' with '1' to get t.

var isOneEditDistance = function(s, t) {
  if (s === t) return false
  let str1 = s.length < t.length ? t : s;
  let str2 = s.length >= t.length ? t : s;
  let c = 0;
  let p = 0;
  
  for (let i=0;i<str1.length;i++) {
      if (str1[i] !== str2[p]) {
          c++
          if (str1.length === str2.length) {
              p++
          } 
          if (str1[i+1] === str2[p]) {
              i++
              p++
          }
      } else {
          console.log('line 27')
          p++
      }
  }
  return c <= 1 ? true : false
};

File: LeetCode/bi_search_and_2_point.py/one_edit_distance.py
class Solution:
    def isOneEditDistance(self, s: str, t: str) -> bool:
        if abs(len(s) - len(t)) > 1 or s == t:
            return False
        
        i = j = 0
        edit = 1
        while i < len(s) and j < len(t):
            if s[i] != t[j]:
                if edit == 0: return False
                edit-=1
                if len(t) > len(s): i-=1
                elif len(s) > len(t): j-=1
            i+=1
            j+=1
        return True
